src/Azure.Functions.Cli/Helpers/KuduLiteDeploymentHelpers.cs

```diff
@@ -58,6 +58,27 @@ public static async Task<DeployStatus> WaitForServerSideBuild(HttpClient client,
             return statusCode;
         }
 
+        public static async Task<DeployStatus> WaitForDedicatedBuildToComplete(HttpClient client, Site functionApp)
+        {
+            // There is a tracked Locking issue in kudulite causing Race conditions, so we have to use this API
+            // to gather deployment progress.
+            ColoredConsole.Write("Remote build in progress, please wait");
+            while (true)
+            {
+                var json = await InvokeRequest<IDictionary<string, bool>>(client, HttpMethod.Get, "/api/isdeploying");
```

Comment:
The retry pattern is added in https://github.com/Azure/azure-functions-core-tools/pull/1436/commits/351e3509fd54b75a7496edfef0dcf61ba231965d#diff-5daf59331ae3b7d5664b9eb70713cf2bR128